addons/im_livechat/controllers/main.py

```diff
@@ -147,6 +147,7 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, cha
             ).sudo().create(channel_vals)
             channel_id = channel.id
             if chatbot_script:
+                channel.livechat_failure = "no_failure"
```

Comment:
Couldn't it be directly provided in `channel_vals`/`_get_livechat_discuss_channel_vals` rather than being a separate write?